### PR TITLE
Add httpse, adblock, and fingerprinting protection switches into Andr…

### PR DESCRIPTION
**Pull request by yrliou**
…oid's privacy page.

Fix https://github.com/brave/brave-browser/issues/4830

Screenshot_20190913-012111

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/android/preferences/brave_pref_service_bridge.cc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#include "brave/build/android/jni_headers/BravePrefServiceBridge_jni.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "chrome/browser/profiles/profile_manager.h"
+
+using base::android::JavaParamRef;
+using brave_shields::ControlType;
+
+namespace {
+
+Profile* GetOriginalProfile() {
```
Comment (bridiver) [inline browser/android/preferences/brave_pref_service_bridge.cc]: @SergeyZhukovsky where is the code you already wrote to handle this stuff?

Comment (SergeyZhukovsky) [reply browser/android/preferences/brave_pref_service_bridge.cc]: We set it via preferences in Java currently and we checked the preferences in OnBeforeURLRequest in chrome_network_delegate.cc, but that code is gone right now, @samartnik is moving it to another place.

**Comment by yrliou** [comment]
Changes are android only and CI is suffering from init problem atm for Android build, I have verified the android builds fine locally, so merging.

File: android/java/org/chromium/chrome/browser/preferences/BravePrefServiceBridge.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.preferences;
+
+import org.chromium.base.ThreadUtils;
+
+public class BravePrefServiceBridge extends PrefServiceBridge {
```
Comment (simonhong) [inline android/java/org/chromium/chrome/browser/preferences/BravePrefServiceBridge.java]: @yrliou I have question about this inheritance.
Why did you inherit `PrefServiceBridge` for `BravePrefServiceBridge`?
We are using two instances for `BravePrefServiceBridge` and `PrefServiceBridge`.
and I think `BravePrefServiceBridge` can work w/o `PrefServiceBridge` impl.

**Pull request merged**